vignettes/intro_to_allotalot.Rmd

```diff
@@ -177,66 +183,108 @@ iris_hd
+Since character strings are vectors, do not forget to exclude braces from the assignment expression.
 
 ```{R}
-y: m: d %<-% Sys.Date()
+ch1 : ch2 : ch3 %<-% 'abc'
+ch1
+ch2
+ch3
+```
+
+### Dates
+We can split a Date into its year, month, and day,
+and assign the parts to variables.
+
+```{r}
+y : m : d %<-% Sys.Date()
 y
 m
 d
 ```
 
-#### other classes
+### `lm` summary
+(TBD)
 
+### Class objects
 *allotalot* includes implementations of `unpack` for atomic objects (numerics, characters, etc.), data frames, date objects, and linear model summaries. However, because `unpack` is a generic function we can define new implementations for custom classes. When defining a new implementation keep in mind the implementation needs to return a list so that values are properly unpacked.
 
-### Consume extra values
-
-In some cases, there are many right-hand side values, but we only want to unpack the first couple values, the last couple values, or both. `summary.lm`, for example, unpacks into 11 values. In practice, unpacking 11 values into 11 different variables is not advised. Fortunately, there is a way to assign a couple values and handle the remaining values.
+## Trailing values: the "everything else" clause
+In some cases, we want the first few elements of a list or vector
+but don't care about the trailing elements.
+The `summary` function of `lm`, for example,
+returns a list of 11 values,
+and we might want only the first few.
+Fortunately, there is a way to capture those first few
+and say "don't worry about everything else".
 
-```{R}
+```{r}
 f <- lm(mpg ~ cyl, data = mtcars)
 
-{fcall: fterms: resids: ...rest} %<-% summary(f)
+{fcall : fterms : resids : ...rest} %<-% summary(f)
 
 fcall
 
 fterms
 
-resids
+head(resids)
+```
+
+Here, `rest` will capture everything else.
 
-# `rest` is not fully printed here to save space, but feel
-# free to run the example and look at `rest`
+```{r}
 str(rest)
 ```
 
-As `...` in an argument list collects extra values passed to a function, a variable prefixed with `...` in an assignment expression collects extra values. If we left off the `...` we would get an error.
+The assignment operator noticed that `...rest` is prefixed with `...`,
+and it created a varable called `rest` for the trailing values of the list.
+If we omitted the "everything else" prefix,
+there would be an error because
+the lengths of the left- and right-hand sides of the assignment would
+be mismatched.
 
 ```{R, error = TRUE}
 {fcall: fterms: resids: rest} %<-% summary(f)
 ```
 
-Note that only one collector variable may be specified at each depth of names. If multiple collectors are specified `%<-%` will raise an error.
+Only one collector variable may be specified at each depth of names. If multiple collectors are specified `%<-%` will raise an error.
 
+## Leading values and middle values
 In addition to collecting trailing values, we can also collect initial values and assign remaining values.
```

Comment:
This is my own fault, would you mind removing "assign" from "and assign remaining values"?